modules/tensor_mechanics/src/materials/FiniteStrainCPSlipRateRes.C

```diff
@@ -0,0 +1,355 @@
+{
+  Real rnorm, rnorm0, rnorm_prev;
+  unsigned int iter = 0;
+  Real step = 1.0;
+
+#ifdef DEBUG
+  std::vector<Real> rnormst(_maxiter+1),slipratest(_maxiter+1);//Use for Debugging
+#endif
+
+  calcResidJacobSlipRate();
+  if (_err_tol)
+    return;
+  rnorm = calcResidNorm();
+  rnorm0 = rnorm;
+
+#ifdef DEBUG
+  rnormst[iter] = rnorm;
+  Real slipratemax = 0.0;
+  for (unsigned int i = 0; i < _nss; ++i)
+    if ( std::abs(_slip_rate[i]) > slipratemax )
+      slipratemax = std::abs(_slip_rate[i]);
+  slipratest[iter] = slipratemax;
+#endif
+
+  while ( rnorm > _rtol * rnorm0 && rnorm0 > _abs_tol && iter < _maxiter )
+  {
+    calcUpdate();
+
+    for (unsigned int i = 0; i < _nss; ++i)
+      _slip_rate[i] -= _resid[i];
+
+    calcResidualSlipRate();
+    if (_err_tol)
+      return;
+    rnorm_prev = rnorm;
+    rnorm = calcResidNorm();
+
+    if (_use_line_search && rnorm > rnorm_prev && !lineSearchUpdateSlipRate(rnorm_prev, _resid))
+    {
+#ifdef DEBUG
+      mooseWarning("FiniteStrainCrystalPLasticity: Failed with line search");
+#endif
+      _err_tol = true;
+      return;
+    }
+
+    calcJacobianSlipRate();
+    iter++;
+
+#ifdef DEBUG
+    slipratemax = 0.0;
+    for (unsigned int i = 0; i < _nss; ++i)
+      if ( std::abs(_slip_rate[i]) > slipratemax )
+        slipratemax = std::abs(_slip_rate[i]);
+    rnormst[iter] = rnorm;
+    slipratest[iter] = slipratemax;
+#endif
+  }
+
+  if ( iter == _maxiter )
+  {
+#ifdef DEBUG
+    mooseWarning("FiniteStrainCPSlipRateRes: NR exceeds maximum iteration " << iter << " " << rnorm);
+#endif
+    _err_tol = true;
+    return;
+  }
+}
+
+void
+FiniteStrainCPSlipRateRes::calcResidJacobSlipRate()
+{
+  calcResidualSlipRate();
+  if (_err_tol)
+    return;
+  calcJacobianSlipRate();
+}
+
+void
+FiniteStrainCPSlipRateRes::calcResidualSlipRate()
+{
+  RankTwoTensor iden, eqv_slip_incr, ce, ee;
+
+  iden.zero();
+  iden.addIa(1.0);
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _slip_incr[i] = _slip_rate[i] * _dt;
+
+  eqv_slip_incr.zero();
+  for (unsigned int i = 0; i < _nss; ++i)
+    eqv_slip_incr += _s0[i] * _slip_incr[i];
+
+  eqv_slip_incr = iden - eqv_slip_incr;
+
+  _fp_inv = _fp_old_inv * eqv_slip_incr;
+  _fe = _dfgrd_tmp * _fp_inv;
+
+  ce = _fe.transpose() * _fe;
+  ee = ce - iden;
+  ee *= 0.5;
+
+  _pk2_tmp = _elasticity_tensor[_qp] * ee;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _tau[i] = _pk2_tmp.doubleContraction(_s0[i]);
+
+  update_slip_system_resistance();
+  getSlipIncrements();
+
+  if (_err_tol)
+    return;
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    _resid[i] = _slip_rate[i] - _slip_incr[i] / _dt;
+}
+
+void
+FiniteStrainCPSlipRateRes::calcJacobianSlipRate()
+{
+  //_dsliprate_dsliprate not reinitialized to zero, hence order is important
+  calcDtauDsliprate();
+  calcDgssDsliprate();
+
+  for (unsigned int i = 0; i < _nss; ++i)
+    for (unsigned int j = 0; j < _nss; ++j)
+    {
+      _jacob[i][j] = 0.0;
+      if ( i == j ) _jacob[i][j] += 1.0;
```

Comment:
we don't put spaces on the interior of round braces (there is one case above as well)